src/pages/Scholarships.jsx

```diff
@@ -29,55 +29,58 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
-  const [amountFilterVals, setAmountFilterVals] = useState({
-    minAmount: 0,
-    maxAmount: 0,
-  });
+
   const location = useLocation();
   const history = useHistory();
 
-  useEffect(() => {
-    const params = queryString.parse(location.search, { parseNumbers: true });
-    const minParam = params.min ?? 0;
-    const maxParam = params.max ?? 0;
-    setAmountFilterVals({
-      minAmount: minParam > 0 ? minParam : 0,
-      maxAmount: maxParam > 0 ? maxParam : 0,
+  // sets min and max amount for the amount filter
+  const setMinMax = (index, val) => {
+    history.push({
+      search: queryString.stringify({
+        ...queryString.parse(location.search),
```

Comment:
Define `params` before this function and use this:
```suggestion
        ...params,
```